### PR TITLE
Corsp Trakign

### PR DESCRIPTION
**Pull request by arctic5**
Corsp Trakign

### CODE REVIEW

**Comment by Medo42** [comment]
Jaja das gud cod
image

File: Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml [hunk 1]
```diff
@@ -77,6 +77,11 @@ else if(tracking)
         }
         else
             stopTracking = true;
+        if (player.corpse and !player.object and tracking)
```
Comment (TazeTSchnitzel) [inline Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml]: this should be `player.corpse != noone`

Comment (Medo42) [inline Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml]: Meep, no points for you either. It should be `instance_exists(player.corpse)`. I provoked the error by setting respawn time to 20 seconds.

Comment (TazeTSchnitzel) [inline Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml]: Oh, right.

Actually, does GM even guarantee that object IDs aren't reused?

**Comment by MrRatermat** [comment]
instance_create() increases a hidden global object id counter. This counter, being an unsigned int, can go into massive numbers before it has to cycle back to 0.

**Comment by TazeTSchnitzel** [comment]
Ah, I see!

...still, GM's approach to this is a little concerning. It'd be so much better if it had true weak references rather than this kludge.

File: Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml (hunk 1 above)
Comment (MrRatermat) [inline Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml]: (Reposted)
instance_create() increases a hidden global object id counter. This counter, being an unsigned int, can go into massive numbers before it has to cycle back to 0.
So no, it doesn't guarantee it (Like all counters), but its REALLY difficult to make it reuse an object ID.

**Comment by Medo42** [comment]
If you insist I could make the DeadGuy reset player.corpse to noone...

**Comment by TazeTSchnitzel** [comment]
It should do that, yes.

**Comment by Medo42** [comment]
Fine, although I think we rely on decayed instance IDs not being quickly re-used in lots of places.

**Comment by Medo42** [comment]
Actually now that I implemented it I don't want to do it, since it opens a new can of worms - the corpse would have to have a reference to the player as well in order to set player.corpse to noone, but then the player can disconnect and that won't remove the corpse from the game and so player needs to also set corpse.owner to noone or something similar... too much hassle to fix something that isn't broken.

**Pull request merged**